state-sync/state-sync-v2/state-sync-driver/src/error.rs

```diff
@@ -16,6 +16,8 @@ pub enum Error {
     BootstrapNotComplete(String),
     #[error("Failed to send callback: {0}")]
     CallbackSendFailed(String),
+    #[error("Timed-out waiting for a data stream too many times.")]
+    CriticalDataStreamTimeout(String),
```

Comment:
Is this just for one peer or all peers?  `Critical` makes it seem like it might not be able to get data from anywhere.

Reply:
Yeah, this is from all peers, so it includes the case where the data might not be available.